src/index.js

```diff
@@ -1,14 +1,36 @@
-const React = require('react');
-const codemirror = require('./codemirror.jsx');
-
-// eslint-disable-next-line react/display-name
-module.exports = (code, lang, opts = { dark: false, tokenizeVariables: false }) =>
-  React.createElement(
-    'span',
-    {
-      className: opts.dark ? 'cm-s-tomorrow-night' : 'cm-s-neo',
-    },
+import React from 'react';
+import codemirror from './codeMirror';
+import codeEditor from './codeEditor';
+import uppercase from './utils/uppercase';
+import canonical from './utils/canonical';
+
+const SyntaxHighlighter = (
+  code,
+  lang,
+  opts = { dark: false, tokenizeVariables: false, editable: false },
+  editorProps = {}
+) => {
+  if (opts.editable) {
+    return React.createElement(codeEditor, {
+      ...editorProps,
+      code,
+      lang,
+    });
+  }
+
+  let classes = '';
+  if (opts.highlightMode) {
+    classes = 'CodeEditor cm-s-material-palenight';
+  } else {
+    classes = opts.dark ? 'cm-s-tomorrow-night' : 'cm-s-neo';
+  }
```

Comment:
I think we should make a decision between these two dark themes regardless of rendering strategy. (I would personally vote for Material Palenight!) Thoughts?

Reply:
We should, I think this should be bundled into a large decision/work to consolidate the css between this, the `ui`, and `readme` libs.